brightchain-lib/src/lib/interfaces/network/node.ts

```
@@ -0,0 +1,160 @@
+  location: NodeLocation;
+  resources: NodeResources;
+  config: NodeConfig;
+
+  // Network operations
+  connect(): Promise<void>;
+  disconnect(): Promise<void>;
+  updateStatus(status: NodeStatus): Promise<void>;
+
+  // Block operations
+  hasBlock(id: ChecksumBuffer): Promise<boolean>;
+  getBlockMetadata(id: ChecksumBuffer): Promise<any>;
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)